### PR TITLE
Add cron_schedule to StartWorkflowExecution

### PR DESCRIPTION
**Pull request by carlcortright**
We need a way to start cron workflows. This enables this functionality by adding it to the start workflow function in the client.

### CODE REVIEW

File: lib/cadence.rb [hunk 1]
```diff
@@ -27,7 +27,8 @@ def start_workflow(workflow, *input, **args)
         execution_timeout: execution_options.timeouts[:execution],
         task_timeout: execution_options.timeouts[:task],
         workflow_id_reuse_policy: options[:workflow_id_reuse_policy],
-        headers: execution_options.headers
+        headers: execution_options.headers,
+        cronSchedule: execution_options.cron_schedule
```
Comment (antstorm) [inline lib/cadence.rb:31]: ^ underscore case here, for consistency :)

File: examples/bin/load_test [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+# Convert integer strings to integers
+input = args.map { |arg| Integer(arg) rescue arg }
+
+while true
+  sleep(1)
+  workflow_id = SecureRandom.uuid
+  run_id = Cadence.start_workflow(workflow_class, *input, options: { workflow_id: workflow_id })
+  Cadence.logger.info "Started workflow: #{run_id} / #{workflow_id}"
+end
+
```
Comment (antstorm) [inline examples/bin/load_test:19]: ^ This file doesn't seem to be related to this PR

File: lib/cadence/execution_options.rb [hunk 3]
```diff
@@ -2,13 +2,15 @@
 
 module Cadence
   class ExecutionOptions
-    attr_reader :name, :domain, :task_list, :retry_policy, :timeouts, :headers
+    attr_reader :name, :domain, :task_list, :retry_policy, :timeouts, :headers,
```
Comment (antstorm) [inline lib/cadence/execution_options.rb:5]: ExecutionOptions are shared between workflows and activities, so `cron_schedule` probably doesn't belong here being workflow-specific concern

Comment (antstorm) [reply lib/cadence/execution_options.rb:5]: You can just reference it directly from the the options provided, similar to `workflow_id_reuse_policy`

**Review by antstorm — CHANGES_REQUESTED** [review]
Thanks, Carl!

Left some inline comments, plus:
- I think it might be a good idea to have a separate API method for scheduling — `schedule_workflow` with `cron_schedule` being a top-level argument. WDYT?
- We'll also need an API for terminating a workflow, since that's the only way to stop a cron workflow

File: examples/bin/load_test (hunk 2 above)
Comment (carlcortright) [reply examples/bin/load_test:19]: Yeah not sure how this got in here... I'll remove it

**Comment by carlcortright** [comment]
@antstorm addressed your comments and also added the `schedule_workflow` method, I think that's a great idea. 

Thinking for termination we can split that into another PR. Thinking that we already have a few ways to cancel workflows through the UI and CLI. Would like to have a way to do it through ruby, but don't want to block this addition on that.

File: spec/unit/lib/cadence_spec.rb [hunk 4]
```diff
@@ -72,6 +72,35 @@ class TestStartWorkflow < Cadence::Workflow
             )
         end
 
+        it 'starts a cron workflow' do
```
Comment (antstorm) [inline spec/unit/lib/cadence_spec.rb:75]: Can you please update the spec with the new `schedule_workflow` call?

File: examples/bin/trigger_cron [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+workflow_class_name, *args = ARGV
+workflow_class = Object.const_get(workflow_class_name)
+workflow_id = SecureRandom.uuid
+
+# Convert integer strings to integers
+input = args.map { |arg| Integer(arg) rescue arg }
+
+run_id = Cadence.schedule_workflow(
+  workflow_class,
+  '* * * * *',
```
Comment (antstorm) [inline examples/bin/trigger_cron:15]: nit: Should we get the schedule string from command line args as well?

**Review by antstorm — APPROVED** [review]
🎉
